### PR TITLE
Create a shared project for source generator helpers

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/SamplesApp.UITests.Generator/CategoryBucketGenerator.cs [hunk 1]
```diff
@@ -56,7 +51,7 @@ where typeSymbol.GetMembers().OfType<IMethodSymbol>().Any(m => m.FindAttributeFl
 			{
 				var builder = new IndentedStringBuilder();
 
-				builder.AppendLineInvariant("using System;");
+				builder.AppendLineIndented("using System;");
```
Comment (Youssef1313) [inline src/SamplesApp/SamplesApp.UITests.Generator/CategoryBucketGenerator.cs:54]: We were using `IndentedStringBuilder` from `Uno.RoslynHelpers` package. Now, we unify this with other generators that are using `IndentedStringBuilder` defined here in this repo.

File: src/SamplesApp/SamplesApp.UITests.Generator/Helper/SourceGenerationHelper.cs [hunk 2]
```diff
@@ -1,23 +0,0 @@
...
-			context.AnalyzerConfigOptions.GlobalOptions.TryGetValue($"build_property.{name}", out var value);
-			return value ?? defaultValue;
-		}
-
-		public static bool TryGetOptionValue(this GeneratorExecutionContext context, AdditionalText textFile, string key, out string? value)
-		{
-			return context.AnalyzerConfigOptions.GetOptions(textFile).TryGetValue(key, out value);
-		}
-	}
-}
```
Comment (Youssef1313) [inline src/SamplesApp/SamplesApp.UITests.Generator/Helper/SourceGenerationHelper.cs:23]: These helpers already exist. They are added in the new shared project and are shared between `Uno.UI.SourceGenerators` and `SamplesApp.UITests.Generator`

File: src/SolutionTemplate/Uno.ProjectTemplates.Dotnet/content/unoapp-prism/.vsconfig [hunk 3]
```diff
@@ -1,48 +1,109 @@
 {
   "version": "1.0",
   "components": [
-    "microsoft.net.runtime.mono.tooling",
-    "microsoft.net.sdk.emscripten",
-    "wasm.tools",
-    "Component.OpenJDK",
-    "Component.Xamarin",
```
Comment (Youssef1313) [inline src/SolutionTemplate/Uno.ProjectTemplates.Dotnet/content/unoapp-prism/.vsconfig:8]: @jeromelaban Everytime I build a solution filter locally, I see these `.vsconfig`s are modified automatically. Do you know if these changes are correct? Why are they made? and if they are not correct, how to stop the build from doing them?

File: src/Uno.UI.RemoteControl.Host/Uno.UI.RemoteControl.Host.csproj [hunk 4]
```diff
@@ -53,5 +53,6 @@
 	</Target>
 
 	<Target Name="GetTargetPath" />
+	<Target Name="GetCopyToPublishDirectoryItems" />
```
Comment (Youssef1313) [inline src/Uno.UI.RemoteControl.Host/Uno.UI.RemoteControl.Host.csproj:56]: @jeromelaban The build fails for me without this (on master branch too). If this change doesn't look correct, let me know how I can resolve the build failures I see locally.

File: src/SolutionTemplate/Uno.ProjectTemplates.Dotnet/content/unoapp-prism/.vsconfig (hunk 3 above)
Comment (jeromelaban) [reply src/SolutionTemplate/Uno.ProjectTemplates.Dotnet/content/unoapp-prism/.vsconfig:8]: That I do not know. I recently noticed that with VS 17.4, could it be related to deprecated feature? Or just simply disambiguation or duplicates based on hierarchy ?

File: src/Uno.UI.RemoteControl.Host/Uno.UI.RemoteControl.Host.csproj (hunk 4 above)
Comment (jeromelaban) [reply src/Uno.UI.RemoteControl.Host/Uno.UI.RemoteControl.Host.csproj:56]: That's fine, I'm not too sure of what causes these to happen regularly.

File: src/SolutionTemplate/Uno.ProjectTemplates.Dotnet/content/unoapp-prism/.vsconfig (hunk 3 above)
Comment (Youssef1313) [reply src/SolutionTemplate/Uno.ProjectTemplates.Dotnet/content/unoapp-prism/.vsconfig:8]: @jeromelaban I'm not sure. But it's very likely related to this PR https://github.com/unoplatform/uno/pull/7823

**Review by jeromelaban — APPROVED** [review]
You'll need to add the shared project to most of the solution filters

**Comment by Youssef1313** [comment]
Sure. I'll do that.

**Comment by Youssef1313** [comment]
@jeromelaban Is this ready to merge?

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
It does not look like it is, there are still build errors :)

**Comment by Youssef1313** [comment]
@jeromelaban Can you restart macOS tests please? Thanks!

**Pull request merged**